src/main/kotlin/org/malachite/estella/services/UserService.kt

```diff
@@ -15,7 +17,11 @@ class UserService(@Autowired private val userRepository: UserRepository) {
     fun getUser(id: Int): User =
         userRepository.findById(id).get()
 
-    fun addUser(user: User): User = userRepository.save(user)
+    fun addUser(user: User): User = try {
+        userRepository.save(user)
+    } catch (ex: DataIntegrityViolationException) {
+        throw UserAlreadyExistsException()
```

Comment:
Skoro przyjeliśmy przynajmniej chwilowo konwencję że serwisy mają rzucać wyjątki i potem mamy handler w kontrolerze to nie lepiej się tego trzymać?

Reply:
Ale to rzuca normalnie wyjątek - tylko wyjątek domenowy. Plus to jest trochę pozostałość faktu, że trochę te kontrollery były zepsute - ogólnie to dalej rzuci wyjątek, tylko przemianowany na bardziej czytelny.